### ISSUE TITLE
Prover error StateCircuit

### ISSUE THREAD

**Issue by pinkiebell** [privacy-scaling-explorations/zkevm-circuits#969]
common zkevm-circuits commit: https://github.com/privacy-scaling-explorations/zkevm-circuits/commit/48b73bf632266e9cd009a685542aabb044ebebbd
branch with block fixtures: https://github.com/privacy-scaling-explorations/zkevm-chain/tree/prover-error-super-ag-native_withdraw-63c082e431c67eec08325c5ce6e178ae719c472b/errors

```
thread 'tokio-runtime-worker' panicked at 'gen_proof: ConstraintSystemFailure
MockProver: Err(
    [
        ConstraintNotSatisfied {
            constraint: Constraint {
                gate: Gate {
                    index: 2624,
                    name: "state circuit constraints",
                },
                index: 3,
                name: "non-first access reads don't change value",
            },
            location: InRegion {
                region: Region {
                    index: 19,
                    name: "state circuit",
                },
                offset: 3159991,
            },
            cell_values: [
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice,
                            index: 4,
                        },
                        rotation: 0,
                    },
                    "0",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice,
                            index: 555,
                        },
                        rotation: 0,
                    },
                    "1",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice {
                                phase: Phase(
                                    1,
                                ),
                            },
                            index: 10,
                        },
                        rotation: -1,
                    },
                    "0",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice {
                                phase: Phase(
                                    1,
                                ),
                            },
                            index: 10,
                        },
                        rotation: 0,
                    },
                    "1",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Fixed,
                            index: 37,
                        },
                        rotation: 0,
                    },
                    "1",
                ),
            ],
        },
        ConstraintNotSatisfied {
            constraint: Constraint {
                gate: Gate {
                    index: 2624,
                    name: "state circuit constraints",
                },
                index: 3,
                name: "non-first access reads don't change value",
            },
            location: InRegion {
                region: Region {
                    index: 19,
                    name: "state circuit",
                },
                offset: 3160255,
            },
            cell_values: [
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice,
                            index: 4,
                        },
                        rotation: 0,
                    },
                    "0",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice,
                            index: 555,
                        },
                        rotation: 0,
                    },
                    "1",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice {
                                phase: Phase(
                                    1,
                                ),
                            },
                            index: 10,
                        },
                        rotation: -1,
                    },
                    "0",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice {
                                phase: Phase(
                                    1,
                                ),
                            },
                            index: 10,
                        },
                        rotation: 0,
                    },
                    "1",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Fixed,
                            index: 37,
                        },
                        rotation: 0,
                    },
                    "1",
                ),
            ],
        },
        ConstraintNotSatisfied {
            constraint: Constraint {
                gate: Gate {
                    index: 2624,
                    name: "state circuit constraints",
                },
                index: 3,
                name: "non-first access reads don't change value",
            },
            location: InRegion {
                region: Region {
                    index: 19,
                    name: "state circuit",
                },
                offset: 3160519,
            },
            cell_values: [
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice,
                            index: 4,
                        },
                        rotation: 0,
                    },
                    "0",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice,
                            index: 555,
                        },
                        rotation: 0,
                    },
                    "1",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice {
                                phase: Phase(
                                    1,
                                ),
                            },
                            index: 10,
                        },
                        rotation: -1,
                    },
                    "0",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice {
                                phase: Phase(
                                    1,
                                ),
                            },
                            index: 10,
                        },
                        rotation: 0,
                    },
                    "1",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Fixed,
                            index: 37,
                        },
                        rotation: 0,
                    },
                    "1",
                ),
            ],
        },
        ConstraintNotSatisfied {
            constraint: Constraint {
                gate: Gate {
                    index: 2624,
                    name: "state circuit constraints",
                },
                index: 3,
                name: "non-first access reads don't change value",
            },
            location: InRegion {
                region: Region {
                    index: 19,
                    name: "state circuit",
                },
                offset: 3160784,
            },
            cell_values: [
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice,
                            index: 4,
                        },
                        rotation: 0,
                    },
                    "0",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice,
                            index: 555,
                        },
                        rotation: 0,
                    },
                    "1",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice {
                                phase: Phase(
                                    1,
                                ),
                            },
                            index: 10,
                        },
                        rotation: -1,
                    },
                    "0",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Advice {
                                phase: Phase(
                                    1,
                                ),
                            },
                            index: 10,
                        },
                        rotation: 0,
                    },
                    "1",
                ),
                (
                    VirtualCell {
                        name: "",
                        column: Column {
                            column_type: Fixed,
                            index: 37,
                        },
                        rotation: 0,
                    },
                    "1",
                ),
            ],
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 24,
                    name: "Execution step",
                },
                offset: 148,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 35,
                    name: "Execution step",
                },
                offset: 148,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 24,
                    name: "Execution step",
                },
                offset: 1000,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 35,
                    name: "Execution step",
                },
                offset: 1000,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 24,
                    name: "Execution step",
                },
                offset: 2529,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 35,
                    name: "Execution step",
                },
                offset: 2529,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 24,
                    name: "Execution step",
                },
                offset: 3381,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 35,
                    name: "Execution step",
                },
                offset: 3381,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 24,
                    name: "Execution step",
                },
                offset: 4910,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 35,
                    name: "Execution step",
                },
                offset: 4910,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 24,
                    name: "Execution step",
                },
                offset: 5762,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 35,
                    name: "Execution step",
                },
                offset: 5762,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 24,
                    name: "Execution step",
                },
                offset: 7291,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 35,
                    name: "Execution step",
                },
                offset: 7291,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 24,
                    name: "Execution step",
                },
                offset: 8143,
            },
        },
        Lookup {
            name: "Rw",
            lookup_index: 339,
            location: InRegion {
                region: Region {
                    index: 35,
                    name: "Execution step",
                },
                offset: 8143,
            },
        },
    ],
)', /app/prover/src/utils.rs:65:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[2022-12-05T18:46:08Z INFO  prover::shared_state] task_result: Err(
        "gen_proof: ConstraintSystemFailure\nMockProver: Err(\n    [\n        ConstraintNotSatisfied {\n            constraint: Constraint {\n                gate: Gate {\n                    index: 2624,\n                    name: \"state circuit constraints\",\n                },\n                index: 3,\n                name: \"non-first access reads don't change value\",\n            },\n            location: InRegion {\n                region: Region {\n                    index: 19,\n                    name: \"state circuit\",\n                },\n                offset: 3159991,\n            },\n            cell_values: [\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice,\n                            index: 4,\n                        },\n                        rotation: 0,\n                    },\n                    \"0\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice,\n                            index: 555,\n                        },\n                        rotation: 0,\n                    },\n                    \"1\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice {\n                                phase: Phase(\n                                    1,\n                                ),\n                            },\n                            index: 10,\n                        },\n                        rotation: -1,\n                    },\n                    \"0\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice {\n                                phase: Phase(\n                                    1,\n                                ),\n                            },\n                            index: 10,\n                        },\n                        rotation: 0,\n                    },\n                    \"1\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Fixed,\n                            index: 37,\n                        },\n                        rotation: 0,\n                    },\n                    \"1\",\n                ),\n            ],\n        },\n        ConstraintNotSatisfied {\n            constraint: Constraint {\n                gate: Gate {\n                    index: 2624,\n                    name: \"state circuit constraints\",\n                },\n                index: 3,\n                name: \"non-first access reads don't change value\",\n            },\n            location: InRegion {\n                region: Region {\n                    index: 19,\n                    name: \"state circuit\",\n                },\n                offset: 3160255,\n            },\n            cell_values: [\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice,\n                            index: 4,\n                        },\n                        rotation: 0,\n                    },\n                    \"0\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice,\n                            index: 555,\n                        },\n                        rotation: 0,\n                    },\n                    \"1\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice {\n                                phase: Phase(\n                                    1,\n                                ),\n                            },\n                            index: 10,\n                        },\n                        rotation: -1,\n                    },\n                    \"0\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice {\n                                phase: Phase(\n                                    1,\n                                ),\n                            },\n                            index: 10,\n                        },\n                        rotation: 0,\n                    },\n                    \"1\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Fixed,\n                            index: 37,\n                        },\n                        rotation: 0,\n                    },\n                    \"1\",\n                ),\n            ],\n        },\n        ConstraintNotSatisfied {\n            constraint: Constraint {\n                gate: Gate {\n                    index: 2624,\n                    name: \"state circuit constraints\",\n                },\n                index: 3,\n                name: \"non-first access reads don't change value\",\n            },\n            location: InRegion {\n                region: Region {\n                    index: 19,\n                    name: \"state circuit\",\n                },\n                offset: 3160519,\n            },\n            cell_values: [\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice,\n                            index: 4,\n                        },\n                        rotation: 0,\n                    },\n                    \"0\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice,\n                            index: 555,\n                        },\n                        rotation: 0,\n                    },\n                    \"1\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice {\n                                phase: Phase(\n                                    1,\n                                ),\n                            },\n                            index: 10,\n                        },\n                        rotation: -1,\n                    },\n                    \"0\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice {\n                                phase: Phase(\n                                    1,\n                                ),\n                            },\n                            index: 10,\n                        },\n                        rotation: 0,\n                    },\n                    \"1\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Fixed,\n                            index: 37,\n                        },\n                        rotation: 0,\n                    },\n                    \"1\",\n                ),\n            ],\n        },\n        ConstraintNotSatisfied {\n            constraint: Constraint {\n                gate: Gate {\n                    index: 2624,\n                    name: \"state circuit constraints\",\n                },\n                index: 3,\n                name: \"non-first access reads don't change value\",\n            },\n            location: InRegion {\n                region: Region {\n                    index: 19,\n                    name: \"state circuit\",\n                },\n                offset: 3160784,\n            },\n            cell_values: [\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice,\n                            index: 4,\n                        },\n                        rotation: 0,\n                    },\n                    \"0\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice,\n                            index: 555,\n                        },\n                        rotation: 0,\n                    },\n                    \"1\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice {\n                                phase: Phase(\n                                    1,\n                                ),\n                            },\n                            index: 10,\n                        },\n                        rotation: -1,\n                    },\n                    \"0\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Advice {\n                                phase: Phase(\n                                    1,\n                                ),\n                            },\n                            index: 10,\n                        },\n                        rotation: 0,\n                    },\n                    \"1\",\n                ),\n                (\n                    VirtualCell {\n                        name: \"\",\n                        column: Column {\n                            column_type: Fixed,\n                            index: 37,\n                        },\n                        rotation: 0,\n                    },\n                    \"1\",\n                ),\n            ],\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 24,\n                    name: \"Execution step\",\n                },\n                offset: 148,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 35,\n                    name: \"Execution step\",\n                },\n                offset: 148,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 24,\n                    name: \"Execution step\",\n                },\n                offset: 1000,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 35,\n                    name: \"Execution step\",\n                },\n                offset: 1000,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 24,\n                    name: \"Execution step\",\n                },\n                offset: 2529,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 35,\n                    name: \"Execution step\",\n                },\n                offset: 2529,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 24,\n                    name: \"Execution step\",\n                },\n                offset: 3381,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 35,\n                    name: \"Execution step\",\n                },\n                offset: 3381,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 24,\n                    name: \"Execution step\",\n                },\n                offset: 4910,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 35,\n                    name: \"Execution step\",\n                },\n                offset: 4910,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 24,\n                    name: \"Execution step\",\n                },\n                offset: 5762,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 35,\n                    name: \"Execution step\",\n                },\n                offset: 5762,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 24,\n                    name: \"Execution step\",\n                },\n                offset: 7291,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 35,\n                    name: \"Execution step\",\n                },\n                offset: 7291,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 24,\n                    name: \"Execution step\",\n                },\n                offset: 8143,\n            },\n        },\n        Lookup {\n            name: \"Rw\",\n            lookup_index: 339,\n            location: InRegion {\n                region: Region {\n                    index: 35,\n                    name: \"Execution step\",\n                },\n                offset: 8143,\n            },\n        },\n    ],\n)",
    )
```

**Comment by aguzmant103** [comment]
Note: it seems Scroll has another PR that fixes this. It's waiting on the spec repo https://discord.com/channels/1035582816488013904/1035582817368813752/1052930707871256586

**Comment by aguzmant103** [comment]
Reviewing #973
